cores/arduino/stm32/dma.c

```diff
@@ -0,0 +1,360 @@
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL WI6LABS BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+  ******************************************************************************
+  */
+
+#include "stm32_def.h"
+#include "dma.h"
+#include "analog.h"
+
+const DmaMap DmaMap_DMA[] = {
+  #if defined(STM32F0xx) || defined(STM32F1xx) || defined(STM32F3xx) || \
+      defined(STM32L0xx) || defined(STM32L1xx) || defined(STM32L4xx)
+  {ADC1, 1, DMA1_Channel1, DMA1_Channel1_IRQn        , 1},
+  #if defined(STM32F0xx)
+  {ADC1, 1, DMA1_Channel2, DMA1_Ch2_3_DMA2_Ch1_2_IRQn, 2},
+  #if defined(STM32F091xC) || defined(STM32F098xx)
+  {ADC1, 2, DMA2_Channel5, DMA1_Ch4_7_DMA2_Ch3_5_IRQn, 5},
+  #endif
+  #endif
+  #if defined(STM32F303xC) || defined(STM32F303xE) || defined(STM32F358xx) || \
+      defined(STM32F398xx)
+  {ADC2, 2, DMA2_Channel1, DMA2_Channel1_IRQn, 1},
+  {ADC2, 2, DMA2_Channel3, DMA2_Channel3_IRQn, 3},
+  {ADC3, 2, DMA2_Channel5, DMA2_Channel5_IRQn, 5},
+  {ADC4, 2, DMA2_Channel2, DMA2_Channel2_IRQn, 2},
+  {ADC4, 2, DMA2_Channel4, DMA2_Channel4_IRQn, 4},
+  #endif
+  #if defined(STM32F328xx) || defined(STM32F303x8) || defined(STM32F334x8)
+  {ADC2, 1, DMA1_Channel2, DMA1_Channel2_IRQn, 2},
+  {ADC2, 1, DMA1_Channel4, DMA1_Channel4_IRQn, 4},
+  #endif
+  #if defined(STM32L0xx)
+  {ADC1, 1, DMA1_Channel2, DMA1_Channel2_3_IRQn, 2},
+  #endif
+  #if defined(STM32L4xx)
+  {ADC1, 2, DMA2_Channel3, DMA2_Channel3_IRQn, 3},
+  #if defined(STM32L476xx) || defined(STM32L476xx)
+  {ADC2, 1, DMA1_Channel2, DMA1_Channel2_IRQn, 2},
+  {ADC3, 1, DMA1_Channel3, DMA1_Channel3_IRQn, 3},
+  {ADC2, 2, DMA2_Channel4, DMA2_Channel4_IRQn, 4},
+  {ADC3, 2, DMA2_Channel5, DMA2_Channel5_IRQn, 5},
+  #endif
+  #endif
+  #endif
+  #if defined(STM32F2xx) || defined(STM32F4xx)
+  {ADC1, 2, DMA2_Stream0, DMA2_Stream0_IRQn, 0},
+  {ADC1, 2, DMA2_Stream4, DMA2_Stream4_IRQn, 4},
+  #if defined(STM32F2xx) || defined(STM32F429xx) || defined(STM32F407xx)
+  {ADC2, 2, DMA2_Stream2, DMA2_Stream2_IRQn, 2},
+  {ADC2, 2, DMA2_Stream3, DMA2_Stream3_IRQn, 3},
+  {ADC3, 2, DMA2_Stream0, DMA2_Stream0_IRQn, 0},
+  {ADC3, 2, DMA2_Stream1, DMA2_Stream1_IRQn, 1},
+  #endif
+  #endif
+};
```

Comment:
isn't the table supposed to end with NP entry ?

Reply:
Yes. Done.